src/cryptsetup/cryptsetup-generator.c

```diff
@@ -196,10 +199,17 @@ static int create_disk(
                         return log_error_errno(errno, "Failed to create symlink %s: %m", to);
 
                 free(to);
-                if (!nofail)
-                        to = strjoin(arg_dest, "/cryptsetup.target.requires/", n, NULL);
-                else
-                        to = strjoin(arg_dest, "/cryptsetup.target.wants/", n, NULL);
+                if (!netdev) {
+                        if (!nofail)
+                                to = strjoin(arg_dest, "/cryptsetup.target.requires/", n, NULL);
+                        else
+                                to = strjoin(arg_dest, "/cryptsetup.target.wants/", n, NULL);
+                } else {
+                        if (!nofail)
+                                to = strjoin(arg_dest, "/remote-cryptsetup.target.requires/", n, NULL);
+                        else
+                                to = strjoin(arg_dest, "/remote-cryptsetup.target.wants/", n, NULL);
+                }
                 if (!to)
```

Comment:
We leak memory pointed by ```to``` when ```symlink``` fails. This seems to be an issue in upstream as well.

Reply:
However, I don't see this as a blocker. So overall, LGTM.

Reply:
Fixed. Upstream uses a different code here.